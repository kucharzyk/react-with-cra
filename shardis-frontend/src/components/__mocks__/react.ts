import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
