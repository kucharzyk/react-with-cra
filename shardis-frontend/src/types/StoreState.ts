export default interface StoreState {
  counter: number;
}
