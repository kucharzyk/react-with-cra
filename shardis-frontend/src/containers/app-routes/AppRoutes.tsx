import * as React from 'react';
import { StatelessComponent } from 'react';
import ServerTime from '../../components/server-time/ServerTime';
import Home from '../../components/home/Home';
import About from '../../components/about/About';
import NotFound from '../../components/not-found/NotFound';
import withStyles from 'material-ui/styles/withStyles';
import { WithStyles } from 'material-ui';
import { Theme } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import { Route, Switch } from 'react-router';

const styles = (theme: Theme) => ({
  root: {
    padding: theme.spacing.unit
  },
});

const AppRoutes: StatelessComponent<{} & WithStyles<'root'>> = (props) => (
  <Paper className={props.classes.root}>
    <Switch>
      <Route path={'/'} exact={true} component={Home}/>
      <Route path={'/home'} component={Home}/>
      <Route path={'/about'} component={About}/>
      <Route path={'/time'} component={ServerTime}/>
      <Route component={NotFound}/>
    </Switch>
  </Paper>
);

export default withStyles(styles)<{}>(AppRoutes);
