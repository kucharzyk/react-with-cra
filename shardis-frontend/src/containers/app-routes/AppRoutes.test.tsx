import * as React from 'react';
import { StatelessComponent } from 'react';
import AppRoutes from './AppRoutes';
import createShallow from 'material-ui/test-utils/createShallow';
import toJson from 'enzyme-to-json';
import StoreState from '../../types/StoreState';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router';
import configureStore from 'redux-mock-store';

describe('AppRoutes', () => {

  let WrappedComponent: StatelessComponent;

  beforeEach(() => {
    const mockStore = configureStore<StoreState>();
    const store = mockStore({
      counter: 1
    });

    WrappedComponent = () => (
      <Provider store={store}>
        <StaticRouter location={'/'} context={{}}>
          <AppRoutes/>
        </StaticRouter>
      </Provider>
    );
  });

  it('renders snapshot', () => {
    const shallowComponent = createShallow({untilSelector: 'AppRoutes'})(<WrappedComponent/>);
    expect(toJson(shallowComponent)).toMatchSnapshot();
  });

});
