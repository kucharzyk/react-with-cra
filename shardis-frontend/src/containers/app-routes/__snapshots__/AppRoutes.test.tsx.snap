// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoutes renders correctly 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-shadow2-6 MuiPaper-rounded-3 AppRoutes-root-1"
>
  <div
    className="Home"
  >
    <h2>
      This is Home component
    </h2>
    <hr />
    <div
      className="Counter"
    >
      <h2>
        Count is: 
        1
      </h2>
      <div>
        <button
          className="MuiButtonBase-root-44 MuiButton-root-29"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label-31"
          >
            -
          </span>
          <span
            className="MuiTouchRipple-root-46"
          />
        </button>
        <button
          className="MuiButtonBase-root-44 MuiButton-root-29"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label-31"
          >
            +
          </span>
          <span
            className="MuiTouchRipple-root-46"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
